Components/Flash/Inc/FlashTask.hpp

```diff
@@ -0,0 +1,68 @@
+/**
+ ******************************************************************************
+ * File Name          : FlashTask.hpp
+ * Description        : Flash interface task. Used for data logging and state recovery
+ ******************************************************************************
+*/
+#ifndef SOAR_FLASHTASK_HPP_
+#define SOAR_FLASHTASK_HPP_
+#include "Task.hpp"
+#include "SystemDefines.hpp"
+#include "SystemStorage.hpp"
+#include "SPIFlash.hpp"
+
+/* Macros/Enums ------------------------------------------------------------*/
+constexpr uint16_t MAX_FLASH_TASK_WAIT_TIME_MS = 5000; // The max time to wait for a command before maintenance is checked
+constexpr uint8_t FLASH_OFFSET_WRITES_UPDATE_THRESHOLD = 5; // The number of writes to flash before offsets are updated in flash
```

Comment:
Note: Remember to use Protocol encoding or something similar (maybe just the raw data frame wrapped in COBS) to be able to recover after unaligned offset writes happen

We should add flash maintenance commands to control messages (flash format etc.)